DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/devlink/test_devlink_sct.py

```diff
@@ -0,0 +1,274 @@
+        stream = await get_sct_streams(dev_groups, src_dst_ports, dent_dev, dut_ports[port_indx])
+        streams.update(stream)
+        custome_stream = {f'custome_stream_{dut_ports[port_indx]}': {
+                        'type': 'raw',
+                        'protocol': '802.1Q' if want_vlan else tc_rule['protocol'],
+                        'ip_source': dev_groups[src_dst_ports[0]][0]['name'],
+                        'ip_destination': dev_groups[src_dst_ports[1]][0]['name'],
+                        'srcMac': tc_rules[dut_ports[port_indx]]['original_rule']['filtertype']['src_mac'],
+                        'dstMac': tc_rules[dut_ports[port_indx]]['original_rule']['filtertype']['dst_mac'],
+                        'frameSize': frame_size,
+                        'rate': SCT_MAP['acl_code_3']['exp'] + 2000}
+                         }
+        overwrite_src_dst_stream_fields(custome_stream, tc_rules[dut_ports[port_indx]]['original_rule'], rule_selectors)
+        streams.update(custome_stream)
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(20 if ports_num == 4 else 10)
+
+    # 5.For each stream, verify CPU rate is as expected
+    coroutines_cpu_stat = []
+    coroutines_devlink = []
+    for trap_name, sct in SCT_MAP.items():
+        if trap_name == 'acl_code_3':
+            continue
+        # Due to absence of ability to suspend streams, increase devitation for sct traffic with expected rate around <=200
+        # in case when we send traffic from 4 ports at once
+        deviation = 0.25 if sct['exp'] <= 200 and ports_num == 4 else DEVIATION
+        coroutines_cpu_stat.append(verify_cpu_traps_rate_code_avg(dent_dev, sct['cpu_code'], sct['exp'], deviation=deviation, logs=True))
+        coroutines_devlink.append(verify_devlink_cpu_traps_rate_avg(dent_dev, trap_name, sct['exp'], deviation=deviation, logs=True))
+    await asyncio.gather(*coroutines_cpu_stat)
+    await asyncio.sleep(5)
+    await asyncio.gather(*coroutines_devlink)
+
+    # 6.Add rule created earlier to trap user defined traffic
+    for port_indx in range(ports_num):
+        out = await TcQdisc.add(
+            input_data=[{dev_name: [
+                {'dev': dut_ports[port_indx], 'direction': 'ingress'}]}])
+        err_msg = f'Verify no error on setting Qdist for {dut_ports[port_indx]}.\n{out}'
+        assert not out[0][dev_name]['rc'], err_msg
+
+        out = await TcFilter.add(input_data=[{dev_name: [tc_rules[dut_ports[port_indx]]['tc_rule']]}])
+        assert not out[0][dev_name]['rc'], f'Failed to create tc rule \n{out}'
+
+    # 7. Verify CPU rate for user defined trap
+    await asyncio.gather(*[verify_cpu_traps_rate_code_avg(dent_dev, SCT_MAP['acl_code_3']['cpu_code'], SCT_MAP['acl_code_3']['exp'], logs=True),
+                           verify_devlink_cpu_traps_rate_avg(dent_dev, 'acl_code_3', SCT_MAP['acl_code_3']['exp'], logs=True)])
```

Comment:
`asyncio.gather()` accepts positional arguments, so you don't need to create a list and unpack it right away:
(also both of the lines are a bit too long)
```suggestion
    await asyncio.gather(
        verify_cpu_traps_rate_code_avg(dent_dev, SCT_MAP['acl_code_3']['cpu_code'],
                                       SCT_MAP['acl_code_3']['exp'], logs=True),
        verify_devlink_cpu_traps_rate_avg(dent_dev, 'acl_code_3', SCT_MAP['acl_code_3']['exp'], logs=True))
```

Reply:
Done